code/controllers/subsystem/weather.dm

```diff
@@ -8,24 +8,66 @@ SUBSYSTEM_DEF(weather)
 	var/list/eligible_zlevels = list()
 	var/list/next_hit_by_zlevel = list() //Used by barometers to know when the next storm is coming
 
-/datum/controller/subsystem/weather/fire()
+/datum/controller/subsystem/weather/fire(resumed = FALSE)
 	// process active weather
-	for(var/V in processing)
-		var/datum/weather/our_event = V
-		if(our_event.aesthetic || our_event.stage != MAIN_STAGE)
+	for(var/datum/weather/weather_event in processing)
+		if(IS_WEATHER_AESTHETIC(weather_event.weather_flags) || weather_event.stage != MAIN_STAGE)
 			continue
-		for(var/mob/act_on as anything in GLOB.mob_living_list)
-			if(our_event.can_weather_act(act_on))
-				our_event.weather_act(act_on)
+
+		if(weather_event.currentpart == SSWEATHER_MOBS)
+			if(!resumed)
+				weather_event.current_mobs = GLOB.mob_living_list.Copy()
+			var/list/current_mobs_cache = weather_event.current_mobs // cache for performance
+			while(current_mobs_cache.len)
+				var/mob/living/target = current_mobs_cache[current_mobs_cache.len]
+				current_mobs_cache.len--
+				if(QDELETED(target))
+					continue
```

Comment:
this shouldn't be possible, if we delete a mob it should get removed from your copied list too (as in, you should go muck around in living destroy)

Reply:
I'll remove it then. Pretty sure I c/p'd this code from another proc so maybe I'll check if it's necessary there as well.

Reply:
for clarity the mob being deleted needs to be handled, I just think it should be handled in the mob's destroy, rather then just letting the reference hang like this.

Reply:
So if the weather subsystem lags out and in between ticks a mob gets `QDELETED` but they would still be inside our old copied player list. (since we cached it) The list gets reset every tick so in the next subsystem tick it will use a new player list without the `QDELETED` mob. 

It makes sense to simple skip the mob here and let the list reset. Are you worried about hard deletes or something? Do I need to switch it to `WEAKREFS()`